data/blog/webview2.md

```diff
@@ -0,0 +1,156 @@
+    | Electron         |   ~4s|
+    | WebView2 + NET5 + WPF |   ~3s|
+    | WebView2 + Win32 (C++)|   ~2s|
+1. Processes Count and Memory
+
+    | Technology | # Processes | Total private bytes | Total working set |
+    | ---        |        ---: |                ---: |              ---: |
+    | Electron   |           4 |             78,940K |          226,396K |
+    | WebView2 (C++)    |           7 |             77,748K |          268,248K |
+    | WebView2 WPF (C#)    |           7 |            102,840K |          307,156K |
+1. Write and Read 10,000 4k Files:
+
+    |          | Write files | Read dir | Sequential Read | Concurrent Read |
+    | ---------|------------:|---------:|----------------:|----------------:|
+    | Electron |     6,218 ms |     21 ms |         8,751 ms |         1,059 ms |
+    | WebView2 (C#) |    10,585 ms |     82 ms |        13,470 ms |         5,936 ms |
+    | WPF w/o WebView2 (C#)     |     4,101 ms |      4 ms |         1,848 ms |           448 ms |
+1. Write and Read 10,000 1MB Files:
+
+    |          | Write files | Read dir | Sequential Read | Concurrent Read |
+    | ---------|------------:|---------:|----------------:|----------------:|
+    | Electron |    29,215 ms |     34 ms |        17,518 ms |         4,602 ms |
+    | WebView2 (C#) |    43,250 ms |     98 ms |        86,222 ms |        35,753 ms |
+    | WPF w/o WebView2 (C#)     |    25,729 ms |      6 ms |        29,536 ms |         3,417 ms |
+1. Sequental IPC Round Trip
+
+    |                              | 1,000 / avg | 10,000 / avg |
+    |------------------------------|------------:|-------------:|
+    | Electron (context isolation) | 211.9 ms / 0.21 ms | 2,400 ms / 0.24 ms |
+    | Electron (node integration)  | 165.8 ms / 0.16 ms | 1,316 ms / 0.13 ms |
+    | WebView2 (C#)                     | 612.6 ms / 0.61 ms | 6,075 ms / 0.61 ms |
+    | WebView2 (C++)                    | 529.0 ms / 0.53 ms | 5,141 ms / 0.51 ms |
+1. Parallel Burst IPC Round Trip
+
+    |                              | 1,000 / avg | 10,000 / avg |
+    | -----------------------------|------------:|-------------:|
+    | Electron (context isolation) | 414 ms / 229.8 ms | 2,021 ms /  949.4 ms  |
+    | Electron (node integration)  | 138 ms /  68.1 ms | 1,349 ms /  627.5 ms  |
+    | WebView2 (C#)                     | 604 ms / 332.5 ms | 5,408 ms / 2,713.8 ms |
+    | WebView2 (C++)                    | 497 ms / 258.3 ms | 3,832 ms / 2,157.5 ms |
+
+### A Note About IPC
+
+In Chromium the browser process acts as an IPC broker between sandboxed renderers and the rest of the system.
+While Electron allows unsandboxed render processes, many apps choose to enable the sandbox for added security.
+WebView2 always has the sandbox enabled, so for most Electron and WebView2 apps IPC can heavily impact overall performance.
+
+Even though Electron and WebView2 have a similar process models, their underlying IPC implementation is different.
```

Comment:
I've learned in this section that IPC between Electron and WebView2 are different, but we never explicitly state what underlying IPC implementation WebView2 uses.

Is this implying that WV2 uses the JSON string marshalling and is slower?

Reply:
Yes. The following phrase should be rewritten if that was not clear:

> Communicating between JavaScript and C++ or C# requires marshalling

Reply:
We noted that JSON is most common, and that Edge has some optimizations planned for M93.